checkov/kustomize/runner.py

```diff
@@ -333,65 +336,87 @@ def _get_parsed_output(self, filePath, extractDir, output):
+        try:
+            output = Runner._get_kubectl_output(filePath, templateRendererCommand, kustomizeProcessedFolderAndMeta)
+        except Exception:
+            logging.warning(f"Error building Kustomize output at dir: {filePath}.", exc_info=True)
+            return
+
+        env_or_base_path_prefix = Runner._get_env_or_base_path_prefix(filePath, kustomizeProcessedFolderAndMeta)
+        if env_or_base_path_prefix is None:
+            logging.warning(f"env_or_base_path_prefix is None, filePath: {filePath}", exc_info=True)
+            return
+
+        extractDir = target_folder_path + env_or_base_path_prefix
+        os.makedirs(extractDir, exist_ok=True)
+
+        logging.debug(f"Kustomize: Temporary directory for {filePath} at {extractDir}")
+        output = str(output, 'utf-8')
+        cur_writer = Runner._get_parsed_output(filePath, extractDir, output, sharedKustomizeFileMappings)
+        if cur_writer:
+            Runner._curWriterValidateStoreMapAndClose(cur_writer, filePath, sharedKustomizeFileMappings)
+
     def run_kustomize_to_k8s(self, root_folder, files, runner_filter):
         kustomizeDirectories = self._findKustomizeDirectories(root_folder, files, runner_filter.excluded_paths)
         for kustomizedir in kustomizeDirectories:
             self.kustomizeProcessedFolderAndMeta[kustomizedir] = self._parseKustomization(kustomizedir)
         self.target_folder_path = tempfile.mkdtemp()
         for filePath in self.kustomizeProcessedFolderAndMeta:    
-            logging.debug(f"Kustomization at {filePath} likley a {self.kustomizeProcessedFolderAndMeta[filePath]['type']}")
             if self.kustomizeProcessedFolderAndMeta[filePath]['type'] == 'overlay':
                 self._handle_overlay_case(filePath)
-            try:
-                output = self._get_kubectl_output(filePath)
-            except Exception:
-                logging.warning(f"Error building Kustomize output at dir: {filePath}.", exc_info=True)
-                continue
-
-            env_or_base_path_prefix = self._get_env_or_base_path_prefix(filePath)
-            if env_or_base_path_prefix is None:
-                logging.warning(f"env_or_base_path_prefix is None, filePath: {filePath}", exc_info=True)
-                continue
-
-            extractDir = self.target_folder_path + env_or_base_path_prefix
-            os.makedirs(extractDir, exist_ok=True)
-
-            logging.debug(f"Kustomize: Temporary directory for {filePath} at {extractDir}")
-            output = str(output, 'utf-8')
-            cur_writer = self._get_parsed_output(filePath, extractDir, output)
-            if cur_writer:
-                self._curWriterValidateStoreMapAndClose(cur_writer, filePath)
+        
+        manager = multiprocessing.Manager()
+        # make sure we have new dict
+        sharedKustomizeFileMappings = copy.copy(manager.dict())
```

Comment:
If the dictionary is not shallow (have nested attributes), please use `copy.deepcopy`

Reply:
im clear it anyway but ok